get-resource.sh

```diff
@@ -21,13 +21,33 @@ curl_with_flags() {
     curl "$@"
 }
 
+EXTENSIONS=${EXTENSIONS:-".tar.gz .tar.gz.sha256"}
+
+get_filename_without_extension() {
+	local filename=$1
+	local filename_without_extension=""
+	for ex in ${EXTENSIONS}; do
+		bname=$(basename "${filename}" "${ex}")
+		if [ "${filename:${#bname}}" = "${ex}" ]; then
+			filename_without_extension="${bname}"
+			break
+		fi
+	done
+
+	# If no match is found, fall back to regex
+	if [ "${filename_without_extension}" = "" ]; then
+		filename_without_extension="${filename%.*.*}"
```

Comment:
You can put this as default on L28, and don't need to have the whole `if` statement.